src/Orm/BaseModel/index.ts

```diff
@@ -1886,6 +1886,17 @@ export class BaseModel implements LucidRow {
        * to the relationships
        */
       const relationOptions = cherryPick ? cherryPick[relation.serializeAs] : undefined
+
+      if (typeof relation.serialize === 'function') {
+        if (Array.isArray(value)) {
+          result[relation.serializeAs] = value.map((one) => one.serialize(relationOptions))
```

Comment:
I think instead of `one.serialize`, we should be calling `relation.serialize`? Maybe, we can have a test for this with a `hasMany` relationship implementation